### DISCUSSION TITLE
Random colors for loose mesh

### DISCUSSION THREAD

**Discussion by Psi25** [bnpr/Malt#413] [Q&A] [answered]
I have this single object with 2 loose mesh, and I want to randomize the colors of this loose meshes. I know there's a Random node but there's no option for randomized colors for loose meshes.

Edit: Cycles has this thing in the Geometry node called Random Per Island. Is there similar in Malt?

![cubes](https://user-images.githubusercontent.com/103315144/193435259-c968633a-c439-46fa-a41b-55d84b9fbc43.png)

**Comment by pragma37** [comment, marked as answer]
Nope, this can't actually be computed inside the shader.
But you can replicate it using geometry nodes:
https://blender.stackexchange.com/questions/258449/is-there-an-alternative-to-random-per-island-for-eevee

In the case of Malt, you will have to store the island index inside a vertex color, since Malt doesn't support other types of mesh attributes.
